sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -90,7 +90,24 @@ trait InvokeLike extends Expression with NonSQLExpression with ImplicitCastInput
     // serializability, because the type-level info with java.io.Serializable and
     // java.io.Externalizable marker interfaces are not strong guarantees.
     // This restriction can be relaxed in the future to expose more optimizations.
-    !dt.existsRecursively(_.isInstanceOf[ObjectType])
+    !dt.existsRecursively {
+      case ObjectType(cls) if cls == classOf[Array[java.lang.Boolean]] => false
```

Comment:
When we use `Invoke` to implement `ToJavaArray`, we must update here, otherwise we cannot hit `ConstantFolding` optimization

Reply:
- `case ObjectType(cls) if cls == classOf[Array[java.lang....]]` must be added here. 
- `case ObjectType(cls) if cls == classOf[java.lang....]` was added by me casually (unrelated to this PR, but I think it should be added)